### ISSUE TITLE
Unable to retrieve download URL when running as a cronjob

### ISSUE THREAD

**Issue by umatz** [mrworf/plexupdate#157]
When running plexupdate as a cronjob, I get the following lines in the syslog:
`Jan  1 06:47:33 ferguson plexupdate: Retrieving list of available distributions`
`Jan  1 06:47:34 ferguson plexupdate: ERROR: Unable to retrieve the URL needed for download (Query DISTRO: ubuntu, BUILD: linux-ubuntu-x86_64)`

If I run the same command directly, the result is better:
`admin2@ferguson:/usr/local/lib/plexupdate$ sudo bash /usr/local/lib/plexupdate/plexupdate.sh --config=/etc/plexupdate/plexupdate.conf`
`Retrieving list of available distributions`
`Your OS reports the latest version of Plex (1.3.3.3148-b38628e) is already installed. Use -f to force download.`

Any idea on what difference in the environment between me running the job with sudo and running directly under root as a cronjob might be the cause?

**Comment by umatz** [comment]
By the wy, I am running plexupdate on ubuntu, so the DISTRO seems to be correct.

**Comment by demonbane** [comment]
Did this just happen once or multiple times? The most likely problem was just a temporary error from the server at the particular time that your cron job ran, and by the time you ran it manually later, it had been resolved.

I've looked at the code for both cronwrapper and plexupdate, and I can't find anything that would cause it to behave differently. cronwrapper literally just runs `plexupdate.sh --config=<your config>` which is exactly what you ran. And plexupdate has no visibility into whether it's being run by cron or not.

The only difference in the environment when running under cron is that you have a very basic PATH, but there are checks at the beginning of the plexupdate run that make sure that all necessary commands are in your path, so you would have gotten an error well before we made it as far as retrieving the download URL.

So my best suggestion is to just wait and see if it ends up happening again. If it was just a temporary server issue it's unlikely to happen again.

P.S. I just checked my logs, and my cron job ran at 6:25 am (UTC -0800) on both the 31st and the 1st, and both times there was no error. And I'm on x86_64 Ubuntu as well.
